pkg/api/alertmanager.go

```diff
@@ -198,6 +198,32 @@ type PostableUserConfig struct {
 	AlertmanagerConfig PostableApiAlertingConfig `yaml:"alertmanager_config" json:"alertmanager_config"`
 }
 
+func (c *PostableUserConfig) UnmarshalJSON(b []byte) error {
```

Comment:
I don't know enough about marshalling and structs but wouldn't this validation be skipped if we did the serialization with YAML? Given we support both, shouldn't the same thing happen on all of them? There's plenty of other structs here that suffer from the same problem.

Reply:
cc: @owen-d given you probably know a bit more than me.

Reply:
We don't support both. Grafana expects JSON and [sends](https://github.com/grafana/grafana/blob/458681c4242022e802d1c1bac4358647778106b6/pkg/services/ngalert/api/lotex_am.go#L150) YAML to the alertmanager.